pkgs/tools/text/ucg/default.nix

```diff
@@ -22,6 +19,20 @@ in stdenv.mkDerivation {
   nativeBuildInputs = [ autoreconfHook pkg-config ];
   buildInputs = [ pcre ];
 
+  installCheckPhase = ''
+    runHook preInstallCheck
+
+    testFile=$(mktemp /tmp/ucg-test.XXXX)
+    echo -ne 'Lorem ipsum dolor sit amet\n2.7182818284590' > $testFile
+    $out/bin/ucg 'dolor' $testFile && echo OK || echo fail
```

Comment:
If any of these commands return nonzero, they will take the `||` path and return the result of the `echo fail` (success) and proceed happily. So we need to either remove the `&&`/`||`, or add another mechanism to make the exit code nonzero.

Reply:
Something like `exit -1`?

Reply:
according to the help section of `ucg`:
```
Exit status is 0 if any matches were found, 1 if no matches, 2 or greater on error.
```
so the following should suffice without the echo parts:
```
$out/bin/ucg pattern $testFile
```